### ISSUE TITLE
pnpm refuses to include packages peerDependency list

### ISSUE THREAD

**Issue by VividLemon** [pnpm/pnpm#5028]
### pnpm version:
7.5.0
### Code to reproduce the issue:
include any peerDependency in a package.json -- my case it was bootstrap
### Expected behavior:
To include the peer
### Actual behavior:
Does not
### Additional information:
I've tried searching around for why pnpm doesn't include it's own peerlist. I find a lot of posts about how their installs, that the peer dependencies of their dependencies aren't downloading. However, mine is different as my packages _needs_ a peer dependency. Does pnpm just not care about them? The desired behavior for the library is that it downloads itself bootstrap (for dev mostly), then ultimately it's up to the user of the library to include bootstrap themselves. How do you achieve this behavior without a peer?
 - `node -v` prints: v16.14.2
 - Windows, macOS, or Linux?: win 11

```json
{
  "name": "pnpm-test",
  "version": "0.1.0",
  "private": true,
  "peerDependencies": {
    "bootstrap": "5.x.x"
  }
}
```
Just a simple example like that, pnpm claims that `pnpm i` is up to date, Whereas npm will download it... Only npm behaves like this, yarn is similar to your library, so there must be a reason I don't understand.

Is a solution to include it as a peerDependency for the user to download, then add it again as a dev dependency?

**Comment by zkochan** [comment]
Yes, you should add bootstrap to your devDependencies as well.

Maybe we should change it, I don't know. It is how it worked in all 3 package managers. npm cli has changed the behavior recently.

**Comment by VividLemon** [comment]
> Yes, you should add bootstrap to your devDependencies as well.
> 
> Maybe we should change it, I don't know. It is how it worked in all 3 package managers. npm cli has changed the behavior recently.

Unless you have any other notes, you can close this. Thank you for your help and your contribution to pnpm. I'm not sure, it could be a good idea. After all, it seems redundant to have to state it in multiple places that it needs to be used. I think that a peer dependency could be treated as a dev with some added bonus for libraries. Thanks again and best of luck with the situation in Ukraine.

**Comment by aspirisen** [comment]
I have the same issue, would like to have some mechanism to install peer dependencies via PNPM

**Comment by VividLemon** [comment]
> I have the same issue, would like to have some mechanism to install peer dependencies via PNPM

Like zkochan said, install them as a dev dependency. They wont be included when packaged.

**Comment by aspirisen** [comment]
> Like zkochan said, install them as a dev dependency

yeah, but i do not have control over package.json in my workspace packages, have to use `packageExtensions` in root package.json

**Comment by zkochan** [comment]
Two similar issues were opened, so I guess it makes sense to implement this. At least, when `auto-install-peers` is `true`.

**Comment by lake2** [comment]
> Two similar issues were opened, so I guess it makes sense to implement this. At least, when `auto-install-peers` is `true`.

This will be execellent, since npm v7 will auto install peer dependency.

**Comment by lake2** [comment]
> Two similar issues were opened, so I guess it makes sense to implement this. At least, when `auto-install-peers` is `true`.

I use pnpm v7.9, this issue seems not be fixed.

> When auto-install-peers is set to true, automatically install direct peer dependencies https://github.com/pnpm/pnpm/pull/5067.
> 
> So if your project the next manifest:
> 
> {
>   "dependencies": {
>     "lodash": "^4.17.21"
>   },
>   "peerDependencies": {
>     "react": "^18.2.0"
>   }
> }
> pnpm will install both lodash and react as a regular dependencies. 

It was not work as description above

**Comment by zkochan** [comment]
I have just tried. It works. You probably did not set `auto-install-peers=true`.

**Comment by lake2** [comment]
> I have just tried. It works. You probably did not set `auto-install-peers=true`.

I have create a repo to reproduce, https://github.com/lake2/peer-repo

**Comment by zkochan** [comment]
I don't see any peerDependencies in this package.json: https://github.com/lake2/peer-repo/blob/master/package.json

**Comment by zkochan** [comment]
This issue is not about hoisting peer dependencies of dependencies to the root of node_modules.

peer dependencies are autoninstalled but when they are peer dependencies of dependencies, they are installed in the node_modules of those dependencies. Not in the root.

**Comment by lake2** [comment]
> This issue is not about hoisting peer dependencies of dependencies to the root of node_modules.
> 
> peer dependencies are autoninstalled but when they are peer dependencies of dependencies, they are installed in the node_modules of those dependencies. Not in the root.

Sorry about that, maybe I misundestood. What I want is this https://github.com/npm/rfcs/blob/main/implemented/0025-install-peer-deps.md. Can pnpm sppport that?

**Comment by lake2** [comment]
I accidentally accomplish the same way as npm v7 does with the two flag:

shamefully-hoist = true
auto-install-peers = true

**Comment by VividLemon** [comment]
> I accidentally accomplish the same way as npm v7 does with the two flag:
> 
> shamefully-hoist = true auto-install-peers = true

That's what `auto-install-peers = true` is for. Shamefully hoist should be omitted if possible since it defeats the purpose of using pnpm. You're better off resolving the conflicts and installing the necessary packages. Though, some libraries seem to need shamefully-hoist, (Nuxt3 for example)

**Comment by lake2** [comment]
> This issue is not about hoisting peer dependencies of dependencies to the root of node_modules.
> 
> peer dependencies are autoninstalled but when they are peer dependencies of dependencies, they are installed in the node_modules of those dependencies. Not in the root.

I guess that npm v7 do two thing, so peer dependeny show in the root node_module:
1. auto intall peer dependeny
2. hoist the peer to the root node_modules

**Comment by VividLemon** [comment]
Doing that, it should behave almostidentically to npm. Which again, kind of defeats the point of pnpm.
